File: frontend/src/types/dream.ts
export type DreamSource = "text" | "voice";
export type DreamStatus = "organized" | "image_generated" | "draft" | "failed";

export type AudioFile = {
  name: string;
  duration: string;
  size: string;
  storage: string;
};

export interface DreamRecord {
  id: string;
  title: string;
  raw_text: string;
  organized_text: string;
  image_prompt: string;
  image_url: string;
  keywords: string[];
  emotions: string[];
  scenes: string[];
  source: DreamSource;
  status: DreamStatus;
  created_at: string;
  updated_at: string;
}

export type Dream = DreamRecord & {
  audio?: AudioFile;
};

export interface DreamListItem {
  id: string;
  title: string;
  image_url: string;
  keywords: string[];
  emotions: string[];
  scenes: string[];
  status: DreamStatus;
  created_at: string;
}

export interface DreamListResponse {
  items: DreamListItem[];
  total: number;
  limit: number;
  offset: number;
}

export interface DreamImageResponse {
  id: string;
  image_url: string;
  status: DreamStatus;
}

export interface HealthResponse {
  status: "ok";
  service: string;
  version: string;
}

export interface ApiErrorBody {
  error: { code: string; message: string };
}

File: backend/app/schemas/dream.py
from pydantic import BaseModel, Field


class DreamCreateRequest(BaseModel):
    raw_text: str = Field(..., description="用户原始梦境碎片")
    source: str = Field(default="text", description="text 或 voice")
    generate_image: bool = False


class DreamReorganizeRequest(BaseModel):
    style: str | None = None


class DreamGenerateImageRequest(BaseModel):
    use_mock: bool = False


class DreamRecordResponse(BaseModel):
    id: str
    title: str
    raw_text: str
    organized_text: str
    image_prompt: str
    image_url: str
    keywords: list[str]
    emotions: list[str]
    scenes: list[str]
    source: str
    status: str
    created_at: str
    updated_at: str


class DreamListItem(BaseModel):
    id: str
    title: str
    image_url: str
    keywords: list[str]
    emotions: list[str]
    scenes: list[str]
    status: str
    created_at: str


class DreamListResponse(BaseModel):
    items: list[DreamListItem]
    total: int
    limit: int
    offset: int


class DreamImageResponse(BaseModel):
    id: str
    image_url: str
    status: str

